Plugins/Damage/Damage.cpp

```diff
@@ -207,4 +209,58 @@ void Damage::OnCombatAttack(NWNXLib::API::CNWSCombatRound *pThis, uint8_t attack
     g_plugin->m_OnCombatAttackHook->CallOriginal<void>(pThis, attackNumber);
 }
 
+//--------------------------- Dealing Damage ----------------------------------
+
+ArgumentStack Damage::DealDamage(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+    int vDamage[13];
+    std::bitset<13> positive;
+
+    // read input
+    uint32_t oidSource = Services::Events::ExtractArgument<Types::ObjectID>(args);
+    uint32_t oidTarget = Services::Events::ExtractArgument<Types::ObjectID>(args);
+
+    for (int k = 0; k < 12; k++)
+    {
+        vDamage[k] = Services::Events::ExtractArgument<int32_t>(args);
+        // need to distinguish between no damage dealt, and damage reduced to 0
+        positive[k] = vDamage[k] > 0;
+    }
+    int damagePower = Services::Events::ExtractArgument<int32_t>(args);
+
+    CNWSCreature *pSource = Globals::AppManager()->m_pServerExoApp->GetCreatureByGameObjectID(oidSource);
```

Comment:
Should probably check if `pSource` is valid too

Reply:
OBJECT_INVALID is ok for the source, works fine - it's only ever passed on as parameter, and the relevant functions handle null just fine.

Reply:
Oh, don't mind me then :D